mqtt_client.c

```diff
@@ -1166,7 +1158,8 @@ static int mqtt_message_receive(esp_mqtt_client_handle_t client, int read_poll_t
              */
             read_len = esp_transport_read(t, (char *)buf, 1, read_poll_timeout_ms);
             if (read_len <= 0) {
-                return esp_mqtt_handle_transport_read_error(read_len, client);
+                esp_mqtt_client_dispatch_transport_error(client);
+                return -1;  // Mid-message timeout or connection-close are errors.
```

Comment:
Okay, this makes sense to treat timeouts/EOFs as errors here, but I'd suggest using the previous notation:

```suggestion
                esp_mqtt_handle_transport_read_error(read_len, client);
                return -1;  // Mid-message timeout or connection-close are errors.
```

Reply:
Done.